src/main/java/uk/gov/hmcts/reform/pip/rules/controllers/RootController.java

```diff
@@ -62,4 +72,106 @@
+            .body(rulesService.getCourt(id));
+    }
+
+    /**
+     * This endpoint returns a court list, based on input search.
+     *
+     * <p>It will apply the business rules against the court and requesting user,
+     * to understand whether they should see the requested court.
+     *
+     * <p>If the user does not have permission to see the resource, or the resource is not found,
+     * then a 404 will be returned instead.
+     *
+     * @param input The ID of the court to search for.
+     * @return The court, if found.
+     */
+    @GetMapping("/courtlist/{input}")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, response = String.class, message = "Court has been found"),
+        @ApiResponse(code = 404, response = String.class, message = "Court has not been found")
+    })
+    public ResponseEntity<List<CourtHearings>> getCourtList(
+        @ApiParam(value = "The input search to retrieve", required = true) @PathVariable String input) {
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(rulesService.getCourtList(input));
+    }
+
+    /**
+     * This endpoint returns a court list.
+     *
+     * <p>It will apply the business rules against the court and requesting user,
+     * to understand whether they should see the requested court.
+     *
+     * <p>If the user does not have permission to see the resource, or the resource is not found,
+     * then a 404 will be returned instead.
+     *
+     * @return The list of all courts, if found.
+     */
+    @GetMapping("/courtlistall")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, response = String.class, message = "Court has been found"),
+        @ApiResponse(code = 404, response = String.class, message = "Court has not been found")
+    })
+    public ResponseEntity<List<CourtHearings>> getCourtList() {
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(rulesService.getCourtList(""));
+    }
+
+    /**
+     * This endpoint returns a hearings list, based on court's ID.
+     *
+     * <p>It will apply the business rules against the court and requesting user,
+     * to understand whether they should see the requested court.
+     *
+     * <p>If the user does not have permission to see the resource, or the resource is not found,
+     * then a 404 will be returned instead.
+     *
+     * @param id The ID of the court to search for.
+     * @return The hearings list, if found.
+     */
+    @GetMapping("/hearings/{id}")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, response = String.class, message = "hearings have been found"),
+        @ApiResponse(code = 404, response = String.class, message = "hearings have not been found")
+    })
+    public ResponseEntity<CourtHearings> getHearings(
+        @ApiParam(value = "The court ID to retrieve", required = true) @PathVariable Integer id) throws ParseException {
+
+        SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT);
```

Comment:
this date logic should be kept out the api, try apply SOLID principles to the code base where possible

Reply:
This will be changed at later stage by passing start date and end date from the front end

Reply:
that doesnt explain why you are creating your own dates in here, either your logic is wrong and you are handling dates at the wrong level or you need to use variables
when the data comes in from the front end we shouldnt have to change the backend to "turn it on", it should be ready to accept however we want it to be sent

Reply:
removed for now because it is not longer needed